3 - Rectangular triangle/solution_3.rb

```diff
@@ -32,7 +32,11 @@ def is_pythagorean?
 
   # check if triangle is isosceles && valid
   def is_isosceles?
-    @first_side == @second_side || @second_side == @third_side || @first_side == @third_side && is_triangle?
+    first_check = @first_side == @second_side
```

Comment:
fixed long line in this method too @alexshvedko

Reply:
ok